substrate-node/pallets/pallet-tft-price/src/lib.rs

```diff
@@ -405,4 +407,41 @@ impl<T: Config> Pallet<T> {
             .round()
             .to_num::<u32>()
     }
+
+    // Validates if the given signer is the next block author based on the validators in session
+    // This can be used if an extrinsic should be refunded by the author in the same block
+    // It also requires that the keytype inserted for the offchain workers is the validator key
+    fn is_next_block_author(
+        signer: &Signer<T, <T as Config>::AuthorityId>,
+    ) -> Result<(), Error<T>> {
+        let author = <pallet_authorship::Pallet<T>>::author();
+        let validators = <pallet_session::Pallet<T>>::validators();
+
+        // Sign some arbitrary data in order to get the AccountId, maybe there is another way to do this?
```

Comment:
I looked a bit in the code and I don't think there is. The sign message is actually iterating over the accounts_from_keys and trying to sign the message. The first one that succeeds will be used to sign the message.